packages/langium-cli/src/generator/grammar-serializer.ts

```diff
@@ -26,12 +28,16 @@ export function serializeGrammar(services: LangiumServices, grammars: Grammar[],
     for (let i = 0; i < grammars.length; i++) {
         const grammar = grammars[i];
         if (grammar.name) {
+            const production = (options.mode ?? config.mode) === 'production';
```

Comment:
Suggestion: overwrite `config.mode` with the value from `options.mode`, if present, somewhere at the beginning of the execution, to ensure that we use a consistent setting everywhere (we might use the new option in other places later).